### PR TITLE
CSHARP-4862: Include TLS1.3 in the list of acceptable TLS protocols.

### PR DESCRIPTION
**Pull request by JamesKovacs**
We don't explicitly test different TLS versions as we rely on the underlying TLS libraries to negotiate the protocol. This commit opts into TLS1.3 as an option. I verified that a `netcoreapp2.1` could still connect to an Atlas cluster with TLS1.2 enabled after adding the enum value for TLS1.3 to the flags. The value of the `SslProtocols.Tls13` flag is from:

https://learn.microsoft.com/en-us/dotnet/api/system.security.authentication.sslprotocols?view=net-8.0

### CODE REVIEW

File: src/MongoDB.Driver.Core/Core/Configuration/SslStreamSettings.cs [hunk 1]
```diff
@@ -53,7 +53,9 @@ public class SslStreamSettings
             _checkCertificateRevocation = checkCertificateRevocation.WithDefault(false);
             _clientCertificates = Ensure.IsNotNull(clientCertificates.WithDefault(Enumerable.Empty<X509Certificate>()), "clientCertificates").ToList();
             _clientCertificateSelectionCallback = clientCertificateSelectionCallback.WithDefault(null);
-            _enabledSslProtocols = enabledProtocols.WithDefault(SslProtocols.Tls12 | SslProtocols.Tls11 | SslProtocols.Tls);
+            // SslProtocols.Tls13 not available until netcoreapp3.1 (not in netstandard2.1) and net5.0
+            const SslProtocols SslProtocolsTls13 = (SslProtocols)12288;
+            _enabledSslProtocols = enabledProtocols.WithDefault(SslProtocolsTls13 | SslProtocols.Tls12 | SslProtocols.Tls11 | SslProtocols.Tls);
```
Comment (BorisDog) [inline src/MongoDB.Driver.Core/Core/Configuration/SslStreamSettings.cs:58]: @JamesKovacs Do you want to have this as opt-in option and not enabled by default in `SslSettings.EnabledSslProtocols` ?

**Review by BorisDog — APPROVED** [review]
LGTM

**Review by adelinowona — APPROVED** [review]
LGTM

File: src/MongoDB.Driver.Core/Core/Configuration/SslStreamSettings.cs (hunk 1 above)
Comment (JamesKovacs) [reply src/MongoDB.Driver.Core/Core/Configuration/SslStreamSettings.cs]: Great catch. I forgot that we need it in both `SslStreamSettings` (core) and `SslSettings` (high) due to the core/high split.

**Comment by JamesKovacs** [comment]
I don't like scattering the numeric value for `SslProtocols.Tls13` across multiple classes, but because of the core/high split, I would need to make the constant public - which is an even worse choice given that it already exists in `SslProtocols` in newer framework versions. I choose the lesser evil of duplicating the constant value across core/high and tests. If anyone has a better idea, I'm willing to consider it.

**Comment by BorisDog** [comment]
IMHO duplicating the constant is ok for now, as we can handle this in 3.0.
There is also an option of declaring `internal` constant (internals are visible to MongoDB.Driver)

**Comment by JamesKovacs** [comment]
Good point. I thought that we had InternalsVisibleTo between Core/Core.Tests and Driver/Driver.Tests, but not Core/Driver.Tests. I was wrong and we have it between all 3. (Just not Driver/Core.Tests - which makes sense.) Refactored to use an internal constant `SslStreamSettings.SslProtocolsTls13` until we update to `net5.0` as our minimum TFM.

File: tests/MongoDB.Driver.Tests/SslSettingsTests.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@
 using System.Security.Cryptography.X509Certificates;
 using Xunit;
 using FluentAssertions;
+using MongoDB.Driver.Core.Configuration;
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Tests/SslSettingsTests.cs:25]: minor: sort usings.

**Review by BorisDog — APPROVED** [review]
LGTM (+minor comment)

File: tests/MongoDB.Driver.Tests/SslSettingsTests.cs (hunk 2 above)
Comment (JamesKovacs) [reply tests/MongoDB.Driver.Tests/SslSettingsTests.cs:25]: Done.

**Pull request merged**